hasher.go

```diff
@@ -165,19 +165,32 @@ func (n *Hasher) HashLeaf(leaf []byte) []byte {
 	return h.Sum(res)
 }
 
-// HashNode hashes inner nodes to:
-// minNID || maxNID || hash(NodePrefix || left || right), where left and right are the full
-// left and right child node bytes, including their respective min and max namespace IDs:
-// left = left.Min() || left.Max() || l.Hash().
-func (n *Hasher) HashNode(l, r []byte) []byte {
+// HashNode calculates a namespaced hash of a node using the
+// supplied left and right children. The input values, "left" and "right,"
+// are namespaced hash values with the format "minNID || maxNID || hash."
+// By default, the normal namespace hash calculation is followed,
+// which is "res = min(left.minNID, right.minNID) || max(left.maxNID,
+// right.maxNID) || H(NodePrefix, left,
+// right)". "res" refers to the return value of the HashNode.
+// However, if the "ignoreMaxNs" property of the Hasher is set to true,
```

Comment:
[not blocking] for my learning, what is the purpose of the `ignoreMaxNs` field? Do you think it would help readers to add docs to:

```golang
type Hasher struct {
	baseHasher   hash.Hash
	NamespaceLen namespace.IDSize

        // ignoreMaxNs is an optimization that does ...
	ignoreMaxNs      bool
}
```

Reply:
Disregard, I learned from below. Maybe a variation of this comment would be helpful here:

```
	// The "IgnoreMaxNamespace" flag influences the calculation of the
	// namespace ID range for intermediate nodes in the tree.
	// This flag signals that, when determining the upper limit of the
	// namespace ID range for a tree node, the maximum possible namespace ID
	// (equivalent to "NamespaceIDSize" bytes of 0xFF, or 2^NamespaceIDSize-1)
	// should be omitted if feasible. For a more in-depth understanding of
	// this field, refer to the "HashNode" method in the "Hasher.
```

Reply:
Sure, done https://github.com/celestiaorg/nmt/pull/96/commits/26b1ea676ac264bb2c6894efcd7a4f4aaf3924b3